src/client/common/alias_dict.cpp

```diff
@@ -97,37 +142,166 @@ bool mp::AliasDict::remove_alias(const std::string& alias)
     return false;
 }
 
-std::vector<std::string> mp::AliasDict::remove_aliases_for_instance(const std::string& instance)
+bool mp::AliasDict::remove_context(const std::string& context)
+{
+    if (aliases.erase(context) > 0)
+    {
+        modified = true;
+
+        if (active_context == context)
+            set_active_context(default_context_name);
+
+        return true;
+    }
+
+    return false;
+}
+
+// This function removes the context called as the instance, and then it iterates through all the remaining contexts
+// to see if there are aliases defined for the instance.
+std::vector<std::pair<std::string, std::string>> mp::AliasDict::remove_aliases_for_instance(const std::string& instance)
 {
-    std::vector<std::string> removed_aliases;
+    std::vector<std::pair<std::string, std::string>> removed_aliases;
+
+    remove_context(instance);
+
+    const auto old_context_name = active_context;
 
-    for (auto it = aliases.begin(); it != aliases.end();)
+    for (auto dict_it = begin(); dict_it != end(); ++dict_it)
+        for (auto context_it = dict_it->second.begin(); context_it != dict_it->second.end(); ++context_it)
+            if (context_it->second.instance == instance)
+                removed_aliases.emplace_back(std::make_pair(dict_it->first, context_it->first));
+
+    for (auto pair_to_remove : removed_aliases)
     {
-        if (it->second.instance == instance)
-        {
-            modified = true;
-            removed_aliases.push_back(it->first);
-            it = aliases.erase(it);
-        }
-        else
-            ++it;
+        auto [context_from, removed_alias] = pair_to_remove;
```

Comment:
Idem- can assign the pair in the `for()`.

Reply:
Done.